src/EditorFeatures/Core/Implementation/EditAndContinue/EditAndContinueDiagnosticAnalyzer.cs

```diff
@@ -34,20 +36,16 @@ public override Task<ImmutableArray<Diagnostic>> AnalyzeSyntaxAsync(Document doc
 
         public override Task<ImmutableArray<Diagnostic>> AnalyzeSemanticsAsync(Document document, CancellationToken cancellationToken)
         {
-            var services = document.Project.Solution.Workspace.Services;
-            var encService = services.GetService<IEditAndContinueWorkspaceService>();
-            if (encService is null)
-            {
-                return SpecializedTasks.EmptyImmutableArray<Diagnostic>();
-            }
+            var workspace = document.Project.Solution.Workspace;
+            _lazyProxy ??= new RemoteEditAndContinueServiceProxy(workspace);
```

Comment:
this needs to at least be doc'ed as to if it's ok that we might have two threads instantiate the RemoteEditAndContinueServiceProxy

Reply:
It's an immutable object.